### PR TITLE
(Feat). Implemented class visiting

### PR DESCRIPTION
**Pull request by toman-tom**
Hey @T14Raptor  - can you check you're happy with this commit? It should implement proper visiting for classes

### CODE REVIEW

File: ast/visit.go [hunk 1]
```diff
@@ -996,3 +1018,31 @@ func (n *VariableDeclaration) VisitWith(v Visitor) {
...
 }
+
+func (n *FieldDefinition) VisitWith(v Visitor) {
+	v.VisitFieldDefinition(n)
+}
+
+func (n *FieldDefinition) VisitChildrenWith(v Visitor) {
+	n.Key.VisitWith(v)
+	if n.Initializer != nil {
+		(*n.Initializer).VisitWith(v)
```
Comment (T14Raptor) [inline ast/visit.go]:
```suggestion
		n.Initializer.VisitWith(v)
```

File: ast/visit.go [hunk 2]
```diff
@@ -592,8 +607,15 @@ func (c *ClassLiteral) VisitWith(v Visitor) {
 }
 
 func (c *ClassLiteral) VisitChildrenWith(v Visitor) {
-	c.Name.VisitWith(v)
-	c.SuperClass.VisitWith(v)
+	if c.Name != nil {
+		c.Name.VisitWith(v)
+	}
+	if c.SuperClass != nil {
+		(*c.SuperClass).VisitWith(v)
```
Comment (T14Raptor) [inline ast/visit.go]:
```suggestion
		c.SuperClass.VisitWith(v)
```

File: ast/visit.go [hunk 3]
```diff
@@ -592,8 +607,15 @@ func (c *ClassLiteral) VisitWith(v Visitor) {
...
+	}
+	for _, element := range c.Body {
+		element.(VisitableNode).VisitWith(v)
```
Comment (T14Raptor) [inline ast/visit.go:617]: Should just make the `ClassElement` interface embed `VisitableNode`

**Review by T14Raptor — CHANGES_REQUESTED** [review]

File: ast/visit.go (hunk 3 above)
Comment (toman-tom) [reply ast/visit.go:617]: Resolved in #8

**Pull request merged**